.github/workflows/build_and_test_cosmian_vm.yml

```diff
@@ -0,0 +1,166 @@
+---
+name: Build and Publish VM Image
+
+on:
+  workflow_call:
+    inputs:
+      distrib:
+        required: true
+        type: string
+
+env:
+  GCP_DEV_PROJECT: cosmian-dev
+  GCP_PUBLIC_PROJECT: cosmian-public
+
+jobs:
+  build-kms-cosmian-vm-image:
+    name: Build KMS in Cosmian VM Image for SEV
+    runs-on: ubuntu-22.04
+    outputs:
+      timestamp: ${{ steps.env.outputs.TIMESTAMP }}
+      image_name: ${{ steps.env.outputs.IMAGE_NAME }}
+      ci_instance: ${{ steps.env.outputs.CI_INSTANCE }}
+    permissions:
+      contents: read
+      id-token: write
+    defaults:
+      run:
+        working-directory: ./packer
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Create env variables
+        id: env
+        run: |
+          TIMESTAMP="$(date -u +'%Y%m%d%H%M%S')"
+          echo "TIMESTAMP=$TIMESTAMP" >> "$GITHUB_OUTPUT"
+          echo "IMAGE_NAME=alpha-$TIMESTAMP-cosmian-vm-kms-sev-${{ inputs.distrib }}" >> "$GITHUB_OUTPUT"
+          echo "CI_INSTANCE=gh-ci-cosmian-vm-kms-$TIMESTAMP-${{ inputs.distrib }}" >> "$GITHUB_OUTPUT"
+
+      - name: Setup `packer`
+        uses: hashicorp/setup-packer@main
+
+      - name: Run `packer init`
+        run: packer init -machine-readable gcp-sev-${{ inputs.distrib }}.pkr.hcl
+
+      - name: Install plugins
+        run: |
+          packer plugins install github.com/hashicorp/amazon
+          packer plugins install github.com/hashicorp/googlecompute
+
+      - name: Authenticate to Google Cloud project
+        uses: google-github-actions/auth@v2
+        with:
+          credentials_json: ${{ secrets.GOOGLE_COSMIAN_DEV_CREDENTIALS }}
+
+      - name: Build GCP images - main
+        env:
+          TIMESTAMP: ${{ steps.env.outputs.TIMESTAMP }}
+        if: startsWith(github.ref, 'refs/tags/') != true
+        run: |
+          packer build -var "prefix=alpha-$TIMESTAMP" gcp-sev-${{ inputs.distrib }}.pkr.hcl
+
+      # - name: Build GCP images - release
+      #   if: startsWith(github.ref, 'refs/tags')
+      #   run: |
+      #     packer build -var "prefix=release-${GITHUB_REF_NAME//./-}" gcp-sev-${{ inputs.distrib }}.pkr.hcl
+
```

Comment:
```suggestion
      - name: Build GCP images - release
        if: startsWith(github.ref, 'refs/tags')
        run: |
          packer build -var "prefix=release-${GITHUB_REF_NAME//./-}" gcp-sev-${{ inputs.distrib }}.pkr.hcl

```

Reply:
I will add a better way to release the image with a specific step

Reply:
d47349e1